torch/csrc/autograd/saved_variable.cpp

```diff
@@ -90,6 +98,9 @@ void SavedVariable::save_common_metadata(const Variable& data) {
   }
 }
 
+std::unique_ptr<SavedVariableHooks> SavedVariable::get_default_hooks_() {
+  return Engine::get_default_engine().get_default_saved_variable_hooks();
+}
```

Comment:
I'm not familiar with this piece of code so going to ask a lot of noob questions here...
- Why does this function end in `_`?
- Why does this return a unique_ptr? If we call `get_default_hooks_()` twice, then it returns two different sets of default hooks?
- Should this be a const method? Does this modify the state of SavedVariable?

Reply:
1.  No need for `_`, I will remove them
2. I'm using unique_ptr because it makes it easier to guarantee that the underlying `PyObject* pack_hook` and `unpack_hook` are correctly increfed and decrefed. This does create a new `SavedVariableHooks` object with its own pair of hooks for each `SavedVariable`, but we need that in order for the hooks to survive when default hooks are unset but the `SavedVariable` is still alive. 
As for the current code, it is true that we might be missing out on a optimization by  doing `if (get_default_hooks()) {` instead of something like `if (has_default_hooks()) {`. Do you think it's worth the code duplication?
3. This could indeed be a const method